src/main/java/me/lucien/minesweeper/web/GameController.java

```diff
@@ -29,6 +28,15 @@ public String createRoom(@RequestParam("width") int width, @RequestParam("height
         Room room = new Room(width, height);
         roomMap.put(room.getId(), room);
 
+        Timer timer = new Timer();
```

Comment:
We shouldn't create a timer for a room. Each timer has a background thread and it is wasting resources. Besides, `timer.cancel()` just cancels all the scheduled tasks, while we only indend to cancel one, which makes timer not reusable for multiple rooms. Timer isn't suitable here. ScheduledExecutorService would be better. A thread pool as it is, it's more controllable and it catches errors thrown from tasks.